image_generation/lcm_dreamshaper_v7/cpp/src/main.cpp

```diff
@@ -0,0 +1,312 @@
+        return EXIT_FAILURE;
+    }
+
+    if (result.count("help")) {
+        std::cout << options.help() << std::endl;
+        return EXIT_SUCCESS;
+    }
+
+    std::string positive_prompt = result["posPrompt"].as<std::string>();
+    const std::string device = result["device"].as<std::string>();
+    const uint32_t num_inference_steps = result["step"].as<size_t>();
+    const uint32_t user_seed = result["seed"].as<size_t>();
+    const uint32_t num_images = result["num"].as<size_t>();
+    const uint32_t height = result["height"].as<size_t>();
+    const uint32_t width = result["width"].as<size_t>();
+    const bool use_cache = result["useCache"].as<bool>();
+    const bool read_np_latent = result["readNPLatent"].as<bool>();
+    const std::string model_base_path = result["modelPath"].as<std::string>();
+    const std::string model_type = result["type"].as<std::string>();
+    const std::string lora_path = result["loraPath"].as<std::string>();
+    const float alpha = result["alpha"].as<float>();
+
+    const std::string folder_name = "images";
+    try {
+        std::filesystem::create_directory(folder_name);
+    } catch (const std::exception& e) {
+        std::cerr << "Failed to create dir" << e.what() << std::endl;
+    }
+
+    std::cout << "OpenVINO version: " << ov::get_openvino_version() << std::endl;
+    std::cout << "Running (may take some time) ..." << std::endl;
+
+    // Stable Diffusion pipeline
+
+    StableDiffusionModels models = compile_models(model_base_path + "/" + model_type, device, lora_path, alpha, use_cache);
+    ov::InferRequest unet_infer_request = models.unet.create_infer_request();
+
+    ov::PartialShape sample_shape = models.unet.input("sample").get_partial_shape();
+    OPENVINO_ASSERT(sample_shape.is_dynamic() || (sample_shape[2] * 8 == width && sample_shape[3] * 8 == height),
+        "UNet model has static shapes [1, 4, H/8, W/8] or dynamic shapes [?, 4, ?, ?]");
+
+    // no negative prompt for LCM model: 
+    // https://huggingface.co/docs/diffusers/api/pipelines/latent_consistency_models#diffusers.LatentConsistencyModelPipeline
+    ov::Tensor text_embeddings = text_encoder(models, positive_prompt);
+
+    std::shared_ptr<Scheduler> scheduler = std::make_shared<LCMScheduler>(LCMScheduler(
+        1000, 0.00085f, 0.012f, BetaSchedule::SCALED_LINEAR,
+        PredictionType::EPSILON, {}, 50, true, 10.0f, false,
+        false, 1.0f, 0.995f, 1.0f, read_np_latent));
+    scheduler->set_timesteps(num_inference_steps);
+    std::vector<std::int64_t> timesteps = scheduler->get_timesteps();
+
+    ov::Tensor denoised(ov::element::f32, {1, 4, height / 8, width / 8});
+    for (uint32_t n = 0; n < num_images; n++) {
+        std::uint32_t seed = num_images == 1 ? user_seed: n;
+
+        ov::Tensor latent_model_input = randn_tensor(height, width, read_np_latent, seed);
+
+        float guidance_scale = 8.0;
+        ov::Tensor guidance_scale_embedding = get_w_embedding(guidance_scale, 256);
```

Comment:
`guidance_scale_embedding` creation can be moved outside of the `num_images` loop